common/step_create_cdrom.go

```diff
@@ -167,12 +168,34 @@ var supportedCDISOCreationCommands []cdISOCreationCommand = []cdISOCreationComma
 	},
 }
 
+func isCygwinExecutable(path string) bool {
+	return runtime.GOOS == "windows" && strings.Contains(path, "\\usr\\bin\\")
+}
+
+func toCygwinPath(path string) (string, error) {
+	c := exec.Command("cygpath", path)
```

Comment:
What's the output of this if path is `/c/Users/rgl/AppData/Local/Temp/packer773584472.iso` already?

Reply:
It seems to be the same (but I'm not sure if this is just a coincidence):

```
C:\Users\rgl>C:\tools\msys64\usr\bin\cygpath.exe /c/Users/rgl/AppData/Local/Temp/packer773584472.iso
/c/Users/rgl/AppData/Local/Temp/packer773584472.iso
```

Reply:
I'm not sure if I understood your question, as such, to make sure we are on the same page, this PR is for using packer.exe (installed from packer.io/chocolatey) and xorriso.exe (installed from msys2) in windows (not inside a wsl/wsl2 bash/environment).

So, are you asking me what happens when, inside a wsl/wsl2 bash, we call the native packer.exe (running outside wsl/wsl2 bash/environment) that in turn calls the xorriso that is installed inside of wsl/wsl2? If this was the question, I will gladly test it :-) 

If its something else, can you please describe it a bit more?

Reply:
ah I asked on the wrong part of the code, sorry about that. My question was whether, if a "native" Packer is called from inside a wsl/wsl2 bash environment, isCygwinExecutable() will return true based on the runtime and existence of \usr\bin -- and if so, will it cause problems when executing in that environment?

Supporting the various custom Windows environments has always been a weakness for Packer which we definitely need to get better at.

Reply:
In WSL2 Ubuntu 20.04 (I believe this applies to other "wsl distros"), when a native windows application is executed from wsl, the current $PATH is not propagated, so there is no default way for native packer.exe to pick up any wsl distro binaries (like xorriso); so this MR should not break when running in wsl.

For reference, this is the result of running a native windows application from wsl (as launched by my [rgl/visual-studio-community-vagrant](https://github.com/rgl/visual-studio-community-vagrant) vagrant environment):

```
vagrant@DESKTOP-LIHOA0C:/mnt/c/Users/vagrant$ powershell.exe -Command '$env:PATH -split ";"'
C:\Python38\Scripts\
C:\Python38\
C:\Windows\system32
C:\Windows
C:\Windows\System32\Wbem
C:\Windows\System32\WindowsPowerShell\v1.0\
C:\Windows\System32\OpenSSH\
C:\Program Files\rsync
C:\ProgramData\chocolatey\bin
C:\Program Files\Git\cmd
C:\Program Files (x86)\GitExtensions\
C:\Program Files\Microsoft VS Code\bin
C:\Program Files\docker
C:\Program Files\dotnet\
C:\Program Files\Microsoft SQL Server\130\Tools\Binn\
C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\
C:\VisualStudio2019Community\MSBuild\Current\Bin
C:\Program Files (x86)\dotnet\
C:\Users\vagrant\AppData\Local\Microsoft\WindowsApps

C:\Users\vagrant\AppData\Local\Programs\Fiddler
C:\tools\msys64
C:\Users\vagrant\.dotnet\tools
```

And the was the current wsl $PATH:

```
vagrant@DESKTOP-LIHOA0C:/mnt/c/Users/vagrant$ echo "$PATH" | tr : '\n'
/usr/local/sbin
/usr/local/bin
/usr/sbin
/usr/bin
/sbin
/bin
/usr/games
/usr/local/games
/mnt/c/Python38/Scripts/
/mnt/c/Python38/
/mnt/c/Windows/system32
/mnt/c/Windows
/mnt/c/Windows/System32/Wbem
/mnt/c/Windows/System32/WindowsPowerShell/v1.0/
/mnt/c/Windows/System32/OpenSSH/
/mnt/c/Program Files/rsync
/mnt/c/ProgramData/chocolatey/bin
/mnt/c/Program Files/Git/cmd
/mnt/c/Program Files (x86)/GitExtensions/
/mnt/c/Program Files/Microsoft VS Code/bin
/mnt/c/Program Files/docker
/mnt/c/Program Files/dotnet/
/mnt/c/Program Files/Microsoft SQL Server/130/Tools/Binn/
/mnt/c/Program Files/Microsoft SQL Server/Client SDK/ODBC/170/Tools/Binn/
/mnt/c/VisualStudio2019Community/MSBuild/Current/Bin
/mnt/c/Program Files (x86)/dotnet/
/mnt/c/Users/vagrant/AppData/Local/Microsoft/WindowsApps
/mnt/c/Users/vagrant/AppData/Local/Programs/Fiddler
/mnt/c/tools/msys64
/mnt/c/Users/vagrant/.dotnet/tools
/snap/bin
```

```